pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -583,9 +585,43 @@ self: super:
         propagatedBuildInputs = [ libpciaccess epoxy ] ++ commonPropagatedBuildInputs ++ lib.optionals stdenv.isLinux [
           udev
         ];
-        prePatch = stdenv.lib.optionalString stdenv.hostPlatform.isMusl ''
+        prePatch = (attrs.prePatch or "") + stdenv.lib.optionalString stdenv.hostPlatform.isMusl ''
           export CFLAGS+=" -D__uid_t=uid_t -D__gid_t=gid_t"
         '';
+        patches = [
+          # NVIDIA PRIME Render Offload patchset
+          # Should be upstream by 1.21
+          # http://download.nvidia.com/XFree86/Linux-x86_64/435.17/README/primerenderoffload.html
+          (fetchpatch {
+            url = "https://gitlab.freedesktop.org/xorg/xserver/commit/7f962c70b6d9c346477f23f6c15211e749110078.patch";
+            sha256 = "196nzhw4wipz6a57j1k6wrp4j80z1fq4fpqhw3xcc1mb3yy24pf6";
+          })
+
+          (fetchpatch {
+            url = "https://gitlab.freedesktop.org/xorg/xserver/commit/37a36a6b5b887d5c5a17a6931ceba8ad5d1bb6d5.patch";
+            sha256 = "1q518hwzi7ry4xzydvsghnyb3iz175iwbzqqml1q2s2v5mrc0qjz";
+          })
+
+          (fetchpatch {
+            url = "https://gitlab.freedesktop.org/xorg/xserver/commit/8b67ec7cc6fda243480a5a8ca118b66242f3eb2c.patch";
+            sha256 = "1l29x87vppfwkr2lhijwvypp1ccsfgd9kwx4dk0mskdalspfc2bd";
```

Comment:
hash mismatch in fixed-output derivation '/nix/store/1p2n4apzvn6s66h9l91s0fcssxvg1sma-8b67ec7cc6fda243480a5a8ca118b66242f3eb2c.patch':
  wanted: sha256:1l29x87vppfwkr2lhijwvypp1ccsfgd9kwx4dk0mskdalspfc2bd
  got:    sha256:03h2djzqwl017icxbjjnpxq2zcg14ld78jw60ajnikcy5npdrzqh